plugins/rbac/src/components/CreateRole/PermissionPoliciesFormRow.tsx

```diff
@@ -47,73 +57,109 @@ export const PermissionPoliciesFormRow = ({
-            permissionPoliciesRowData.plugin
-          ]?.permissions ?? []
-        }
-        style={{ width: '450px' }}
-        value={permissionPoliciesRowData.permission ?? ''}
-        onChange={(_e, value) =>
-          onChangePermission(
-            value ?? '',
-            value
-              ? permissionPoliciesData?.pluginsPermissions?.[
-                  permissionPoliciesRowData.plugin
-                ]?.policies?.[value]
-              : undefined,
-          )
-        }
-        getOptionDisabled={getPermissionDisabled}
-        renderInput={(params: any) => (
-          <TextField
-            {...params}
-            label="Permission"
-            name={`${rowName}.permission`}
-            placeholder="Select a permission"
-            error={!!permissionError}
-            helperText={permissionError ?? ''}
-            onBlur={handleBlur}
-            required
-          />
-        )}
-      />
+    <div>
+      <div style={{ display: 'flex', gap: '20px', marginBottom: '20px' }}>
+        <Autocomplete
+          options={permissionPoliciesData?.plugins ?? []}
+          style={{ width: '450px' }}
+          value={permissionPoliciesRowData.plugin ?? ''}
+          onChange={(_e, value) => {
+            onChangePlugin(value ?? '');
+            setSelectedPlugin(value ?? '');
+          }}
+          renderInput={(params: any) => (
+            <TextField
+              {...params}
+              label="Plugin"
+              name={`${rowName}.plugin`}
+              variant="outlined"
+              placeholder="Select a plugin"
+              error={!!pluginError}
+              helperText={pluginError ?? ''}
+              onBlur={handleBlur}
+              required
+            />
+          )}
+        />
+        <Autocomplete
+          disabled={!permissionPoliciesRowData.plugin}
+          options={
+            permissionPoliciesData?.pluginsPermissions?.[
+              permissionPoliciesRowData.plugin
+            ]?.permissions ?? []
+          }
+          style={{ width: '450px' }}
+          value={permissionPoliciesRowData.permission ?? ''}
+          onChange={(_e, value) =>
+            onChangePermission(
+              value ?? '',
+              value
+                ? permissionPoliciesData?.pluginsPermissions?.[
+                    permissionPoliciesRowData.plugin
+                  ]?.policies?.[value]
+                : undefined,
+            )
+          }
+          getOptionDisabled={getPermissionDisabled}
+          renderInput={(params: any) => (
+            <TextField
+              {...params}
+              label="Permission"
+              name={`${rowName}.permission`}
+              variant="outlined"
+              placeholder="Select a permission"
+              error={!!permissionError}
+              helperText={permissionError ?? ''}
+              onBlur={handleBlur}
+              required
+            />
+          )}
+        />
+        <IconButton
```

Comment:
good to comment the CTA now with proper comment as feature is not enabled

Reply:
HI @divyanshiGupta , shall we keep the CTA as well commented for now?

Reply:
@invincibleJai not sure if that is really required. Also if I comment this out I will have to comment out the related tests as well. Lmk if I should do that.

Reply:
Commented out the cta and kept the layout same as before.